testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -555,18 +568,23 @@ impl EventObserver {
         };
         let full_url = format!("http://{url_str}");
 
-        if let Some(db_path) = &self.db_path {
-            let conn =
-                Connection::open(db_path).expect("Failed to open database for event observer");
+        // if the observer is in lossy mode quickly send the payload without checking for the db
+        if self.lossy {
+            Self::send_payload_directly(payload, &full_url, self.timeout, true);
+        } else {
+            if let Some(db_path) = &self.db_path {
+                let conn =
+                    Connection::open(db_path).expect("Failed to open database for event observer");
 
-            // Insert the new payload into the database
-            Self::insert_payload_with_retry(&conn, &full_url, payload, self.timeout);
+                // Insert the new payload into the database
+                Self::insert_payload_with_retry(&conn, &full_url, payload, self.timeout);
 
-            // Process all pending payloads
-            Self::process_pending_payloads(&conn);
-        } else {
-            // No database, just send the payload
-            Self::send_payload_directly(payload, &full_url, self.timeout);
+                // Process all pending payloads
+                Self::process_pending_payloads(&conn);
+            } else {
+                // No database, just send the payload
+                Self::send_payload_directly(payload, &full_url, self.timeout, false);
+            }
```

Comment:
I'm not sure if this is a clippy check, but this should be:

- if `lossy`:
- else if db_path:
- else:

instead of the nested else / if / else